### ISSUE TITLE
Client-side logging

### ISSUE THREAD

**Issue by zenangst** [hyperoslo/ios-foundation#26]
From time to time, it can be very useful to include client-side logging for applications. A really easy way to do this is to setup Sentry (https://github.com/getsentry/sentry-swift). We have this implemented in our own products to trigger alarms for critical parts of the application. I don't know if it can be considered a requirement but I think it should be on everyones radar that this kind of feature exists and is available for anyone who wants to implement it.

When we decided on going with Sentry, we made that decision because the backend also uses the same logging system and it is good for PM's to have it all in one place, we also have our own Sentry server which was also a deciding factor.

With that said, it might not be the defacto answer to what we should use, the bug reports can be a bit wonky at times if you don't set it up correctly and the overview is not super optimal. But it does work and like I said earlier, it is really easy to setup and send "alarms".

**Comment by zenangst** [comment]
Have we followed up on this? Have you guys found yourself adding logging in any of your apps?

**Comment by vadymmarkov** [comment]
@zenangst You've tried Sentry in a couple of projects, right? Do you think it's useful enough to add it in every project by default?

**Comment by zenangst** [comment]
Only if it is deemed necessary, which is a rather vague statement in itself. The reason why I wanted to bring it up was to keep it on peoples radar. The problem with logging is that you don't know that you need it until it is too late. Maybe we could come up with some convention for this to include it in the most crucial parts of any application. Like whenever a `try` method would `throw` and it has a big implication for the user. Again, not really super concrete but... you know 😎 

In on project we monitored `OhMyAuth` requests to alert whenever a refresh token would fail. I think that is an interesting and good example of where you might want to catch and report using Sentry.

**Comment by vadymmarkov** [comment]
Do you have any thought on this @JohnSundell ?

**Comment by onmyway133** [comment]
@zenangst For client side logging, `Sentry` is on the top of my list now

For tracing bugs or tracking predefined important flows in the app, then I would use it. But we should not add this beforehand

**Comment by JohnSundell** [comment]
So with this topic, I assume we're exclusively talking about logging in the term of diagnostics and error reporting? Not about logging for product analytics, right? I've never tried Sentry but would totally be up for trying it. In the past I've used Fabric for this sort of thing, where you can log custom errors. It's nice because it also ties in with the crash reporting from Crashlytics.

**Comment by zenangst** [comment]
@JohnSundell correct, analytics can differ from project to project as the clients might have some pre-existing opinions about that. Sentry is pretty straight forward, we could go for something different but we already have a Sentry instance up and running and the backend guys use that as their service. Sentry is far from perfect but it is pretty easy to setup.

**Comment by JohnSundell** [comment]
Sounds great to me 👍
